ge_worker/bonuses/bonus_processor.py

```diff
@@ -68,28 +93,49 @@ class BonusProcessor:
+        app_context.push()
+
+        # set flask context to show all model data for easier querying using flask models.
+        g.show_all = True
+
+        # get list user objects with corresponding total unique outward transactions and collective unique outward
+        # transactions
+        user_and_total_unique_outward_transactions_list, collective_unique_outward_transactions \
+            = get_list_of_users_with_total_unique_outward_transactions()
+
+        # iterate through list to make disbursements
+        for counter, \
+            (user,
+             unique_outward_transactions_per_user) in enumerate(user_and_total_unique_outward_transactions_list):
+            # compute user's total unique outward transactions as a percentage of the collective unique outward
+            # transactions
             user_percentage_unique_outward_transactions = (
-                (unique_outward_transactions_per_user/total_outward_transactions))
-
-            print('UNIQUE_TX_PER_USR: {}. TOTAL_OUTWARD_TX: {}. USR_PERCENTAGE_TX: {}.'
-                  .format(unique_outward_transactions_per_user,
-                          total_outward_transactions,
-                          user_percentage_unique_outward_transactions))
-
-            bonus_amount = user_percentage_unique_outward_transactions * self.issuable_amount
-
-            # TODO [Philip]: Find a way to avoid this query
-            user = User.query.filter_by(id=user_id).first()
-
-            if bonus_amount > 1 and user:
-                disbursement = make_payment_transfer(bonus_amount,
-                                                     receive_user=user,
-                                                     transfer_subtype=TransferSubTypeEnum.DISBURSEMENT)
-                return disbursement
+                (unique_outward_transactions_per_user / collective_unique_outward_transactions))
+
+            # get user's share of the total issuable amount
+            user_bonus_amount = int(user_percentage_unique_outward_transactions * self.issuable_amount)
+
+            # define logger
+            print('USER: {} {}, '
+                  'USER_TOTAL_UNIQUE_OUTWARD_TX: {}, '
+                  'USER_PERCENTAGE_UNIQUE_OUTWARD_TX: {}, '
+                  'USER_BONUS_AMOUNT: {} '.format(user.first_name,
+                                                  user.last_name,
+                                                  unique_outward_transactions_per_user,
+                                                  user_percentage_unique_outward_transactions,
+                                                  user_bonus_amount))
+
+            if user_bonus_amount > 1:
+                # TODO[Philip]: Results in a synchronous subtask 
+                make_payment_transfer(user_bonus_amount,
+                                      receive_user=user,
+                                      transfer_subtype=TransferSubTypeEnum.DISBURSEMENT)
+
+                db.session.commit()
```

Comment:
@nolash Still wrapping my head around the whole celery dynamic but these changes slightly optimize the daily bonus calculation.
Unfortunately the ```make_payment_transfer``` functions seems to spawn a synchronous task which raises and issues. So that's the next thing I need to figure out

Reply:
@PhilipWafula is `make_payment_transfer` a blocker then? Can you please make an issue for this, describing the exact nature of the problem?

Reply:
Using the format you proposed I imagine ?